content/source/docs/enterprise/saml/identity-provider-configuration-aad.html.md

```diff
@@ -0,0 +1,68 @@
+    - **Source attribute:** (drop-down): user.assignedroles (Custom roles will be created in Azure Active Directory that will be used to map users and groups to TFE teams.)
+    ![Screenshot: Add MemberOf claim](./images/sso-aad-saml-user-claims-memberof.png)
+7. Under the "SAML Signing Certificate" header, download the signing certificate in base64 format
+    ![Screenshot: Download the SAML signing certificate](./images/sso-aad-saml-signing-certificate.png)
+8. Still in the "Single Sign On Configuration" page, under the "Set up <AAD App Name>", copy the following urls that we will enter in our TFE configuration to link TFE to AAD:
+    - **Login URL:**
+    - **Logout URL:**
+    ![Screenshot: Azure Login/Logout URLs](./images/sso-aad-saml-urls.png)
+9. Navigate to `https://<TFE_HOSTNAME>/app/admin/saml` and configure the following:
+    - **Enable SAML single sign-on** (check box): enabled.
+    - **Single Sign-On URL:** Enter the login url from step 8.
+    - **Single Log-out URL:** Enter the logout url from step 8.
+    - **IDP Certificate:** Enter the contents of the PEM (Based64) encoded X.509 certificate captured in step 7.
+
+## Configure Custom Roles for Team Membership Mapping
+
+1. Create teams in TFE as outlined in [TFE Team Membership](./team-membership.html)
+2. Return to the Azure Portal, navigate to "App Registration" page, and search for the application you created for TFE. Select your app and select "Manifest".
+    ![Screenshot: AAD App Registration](./images/sso-aad-saml-app-registration.png)
+3. In the manifest editor, locate the "appRoles" block. This is where you will add additional roles that map users and groups to teams in TFE.
+    ![Screenshot: Manifest Editor](./images/sso-aad-saml-manifest-approles.png)
+4. The "appRoles" block may contain roles automatically generated by AAD. New roles should be added after the system roles and must contain a unique GUID value for the ID value of the new role. You can use a tool such as [GUID Generator](https://www.guidgenerator.com). Click save to add role.
+
+    Example role configuration that creates a new role name Dev:
+    ```{
+    "allowedMemberTypes": [
+        "User"
+    ],
+    "displayName": "Dev",
+    "id": "d1c2ade8-98f8-45fd-aa4a-6d06b947c66f",
+    "isEnabled": true,
+    "description": "Dev Team",
+    "value": "Dev"
+    }```
+    ![Screenshot: New role in Manifest Editor](./images/sso-aad-saml-manifest-devapprole.png)
+5. Back in your application properties, under "Manage" heading, select "Users and Groups". This is where you will enable access to TFE by adding either users or groups to your application. During the process of adding users or groups you will select a role to be assigned to the user or group. Select the role that matches the user or groups TFE team.
```

Comment:
This line is a little confusing - how does the user get back to their application properties? What's meant by "Select the role that matches the user or groups TFE team"?

Reply:
Hmm, should they use the breadcrumb nav at the top? If so, I'd suggest something like, uh...

```suggestion
5. Use the breadcrumb navigation to return to your application's settings. In the left sidebar, under the "Manage" heading, select "Users and Groups". This is where you will enable access to TFE by adding either users or groups to your application. During the process of adding users or groups you will select a role to be assigned to the user or group. Select the role that matches the user or groups TFE team.
```

But I, too, could use a little more clarity about how you select users and groups and how you assign roles.